File: ModAssets/Scripts/lua_env/logger.lua
Log = {}

local function concat_objs(...)
  local r = {}
  for i = 1, select('#', ...) do
    local obj = select(i, ...)
    r[#r + 1] = tostring(obj)
  end
  return table.concat(r, '  ')
end

local DebugColor = ConsoleColor.DarkGray
local WarnColor = ConsoleColor.Yellow
local ErrColor = ConsoleColor.Red
local InfoColor = ConsoleColor.Gray

function Log.debug(...)
  log_print(DebugColor, concat_objs(...))
end

function Log.info(...)
  log_print(InfoColor, concat_objs(...))
end

function Log.warn(...)
  log_print(WarnColor, concat_objs(...))
end

function Log.error(...)
  log_print(ErrColor, concat_objs(...))
end

Logger = Log

File: ModCode/ModMain/Logger.cs
using System;
using System.IO;
using MelonLoader;
using UnityEngine;

namespace MOD_LuaEnv
{
    public static class Logger
    {
        public static bool EnableDebugLog = false;

        // for lua export
        public static void PrintWithColor(ConsoleColor color, string obj) {
            MelonLogger.Msg(color, obj);
        }

        public static void Debug(object obj) {
            if (EnableDebugLog) {
                MelonLogger.Msg($"[LuaEnv] {obj}");
            }
        }

        public static void Info(object obj) {
            MelonLogger.Msg($"[LuaEnv] {obj}");
        }

        public static void Warning(object obj) {
            MelonLogger.Msg(ConsoleColor.Yellow, $"[LuaEnv] {obj}");
        }

        public static void Error(object obj) {
            MelonLogger.Msg(ConsoleColor.Red, $"[LuaEnv] {obj}");
        }
    }
}